kotlinx-coroutines-core/jvm/src/debug/internal/AgentPremain.kt

```diff
@@ -69,3 +69,28 @@ internal object AgentPremain {
         }
     }
 }
+
+/**
+ * This class is loaded if and only if kotlinx-coroutines-core was used as a javaagent argument.
+ * It serves as a "safe" wrapper around [AgentPremain] that does not require any kotlin-stdlib classes to be loaded for its initialization.
+ *
+ * Initialization of [AgentPremain] may fail, e.g., because it started before the required kotlin-stdlib classes were loaded.
+ * If `kotlinx.coroutines.apply.debug.agent` is set to `false`, the coroutine debug agent will not be applied.
+ */
+@Suppress("unused")
+internal object SafeAgentPremain {
+    @JvmStatic
+    fun premain(args: String?, instrumentation: Instrumentation) {
+        // If this property is disabled, the debug agent will not be applied at all
+        val applyDebugAgent = System.getProperty("kotlinx.coroutines.apply.debug.agent", "true").toBoolean()
+        if (!applyDebugAgent) return
```

Comment:
If someone controls the system environment at the point when `premain` is being run, it means they can edit the command line which starts the process, and then, they can simply avoid specifying the Java agent. What's the purpose of adding a new property, then?

Reply:
True, we don't need a property, we can just add the hint to remove the `-javaagent` argument in the error message

Reply:
Removed this extra system property